### DISCUSSION TITLE
mitmproxy ca-certificate not working on Ubuntu

### DISCUSSION THREAD

**Discussion by kleysonr** [mitmproxy/mitmproxy#6340] [Q&A]
I've followed all the instructions to import the mitmproxy ca-certificate in a ubuntu server but looks like it's not working.

Below we can see that the mitmproxy certificatates are being recognized as Not Secure

![notsecure](https://github.com/mitmproxy/mitmproxy/assets/1976289/de47f5d1-adaa-441d-bd7b-4602380d88ee)
 
I imported the ca certificate directly in Chrome

![cert-chrome](https://github.com/mitmproxy/mitmproxy/assets/1976289/72e70450-807d-45d6-940b-9c4612605d06)

And I did the ubuntu configuration as stated in mitm.it/#Linux as well

![cert-linux](https://github.com/mitmproxy/mitmproxy/assets/1976289/5c4f1d79-a18e-449d-82a3-9ec1561dce14)

 "0 added" because it was already added before.

Can someone help me on making it working on ubuntu ?

Best Regards.

**Comment by mhils** [comment]
Do other applications (i.e. not Chrome) work?

**Reply by kleysonr** [reply]
@mhils no.

I have installed Chrome, Edge and Firefox.

None of them are working.

**Comment by kleysonr** [comment]
I found someone complaining about the same issue on this not answerd thread

https://askubuntu.com/questions/1465625/proper-way-to-install-mitmproxys-certificates

Some kind of security level on ubuntu regarding the mitmproxy certifictes ?!?!?!

**Comment by kleysonr** [comment]
I replaced Ubuntu for Fedora and used the steps in mitm.it/Linux#. And my application is not getting Not secure anymore.

**Comment by knightian** [comment]
I used the .pem and converted it to a crt using openssl and that fixed the issue

`openssl x509 -in foo.pem -inform PEM -out foo.crt`

**Comment by robiinkhan** [comment]
@kleysonr have you find any relevant answer?

**Comment by marioeivissa** [comment]
still not working any solution, anybody can help please??

**Comment by MJ-MN** [comment]
### In Chrome
You can import .pem(or .crt) certificate file manually to Chrome
1. Open up `Chrome Settings > Privacy and Security > Security > Manage certificates`. You can also access this part via `chrome://settings/certificates`.
2. Go to `Authorities` tab and import .pem certificate file.

**Comment by dhakadeepak** [comment]
Things that worked for me in Ubuntu 24.04

1. convert the pem file to crt file `openssl x509 -in mitmproxy-ca-cert.pem -inform PEM -out mitmproxy.crt`
2. Place this` mitmproxy.crt`  in `/usr/local/share/ca-certificates`  by doing` sudo cp mitmproxy.crt /usr/local/share/ca-certificates/extra/mitmproxy.crt`
3. then run `sudo update-ca-certificates` 
4. run  `sudo apt install libnss3-tools`
5. run  `certutil -d sql:$HOME/.pki/nssdb -A -t "C,," -n mitmproxy -i /usr/local/share/ca-certificates/mitmproxy.crt`
